WebContent/js/components/dialogs/datasets/DatasetSaveAsDialog.js

```diff
@@ -49,20 +52,38 @@ export default class DatasetSaveAsDialog extends React.Component {
         this.setState({
             newDSName: newValue,
         });
+        // disable the Submit, when DS name is invalid
+        const found = validateName('dataset', newValue);
+        if (found != null && found[0] === newValue && newValue.length <= 44) {
+            this.state.disableSubmit = false;
+            this.state.warning = '';
+        } else {
+            this.state.disableSubmit = true;
+            this.state.warning = ' (WARNING: Invalid Name)';
+        }
     }
 
     updateMember(newValue) {
         this.setState({
             newDSMember: newValue,
         });
+        // disable the Submit, when PDS member name is invalid
+        const found = validateName('datasetMember', newValue);
+        if (found != null && found[0] === newValue) {
```

Comment:
If this is just PDS member, shouldn't length limit here be 8?

Reply:
Or maybe this is covered under your regex expression

Reply:
It is covered under regex expression 
case 'datasetMember':
            regex = /^([A-Z#@$][A-Z0-9#@$-]{0,7})/g;